### PR TITLE
[Keyboard] Valor Rev 2

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Valor Rev 2
This targets develop as it uses STM32L422

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add Valor Rev 2

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xelus/valor/rev2/rev2.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/xelus/valor/rev2/rev2.c:75]: Not needed. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/xelus/valor/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L422
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev2/rules.mk:2]:
```suggestion
MCU = STM32L422

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/xelus/valor/rev2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+OPT = 2
+
+EEPROM_DRIVER = i2c
+
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+RGB_MATRIX_CUSTOM_KB = yes
+
+WS2812_DRIVER = pwm
```
Comment (fauxpark) [inline keyboards/xelus/valor/rev2/rules.mk:32]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
RGB_MATRIX_CUSTOM_KB = yes
WS2812_DRIVER = pwm
EEPROM_DRIVER = i2c

OPT = 2
LTO_ENABLE = yes

LAYOUTS = alice_split_bs
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**